Java_8_to_17/homework/jmp-cloud-service-impl/src/main/java/com/jmp/cloud/service/impl/CloudService.java

```diff
@@ -0,0 +1,41 @@
+package com.jmp.cloud.service.impl;
+
+import com.jmp.dto.BankCard;
+import com.jmp.dto.Subscription;
+import com.jmp.dto.User;
+import com.jmp.service.api.Service;
+
+import java.time.LocalDate;
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class CloudService implements Service {
+    private final HashSet<User> users = new HashSet<>();
+    private final HashSet<Subscription> subscriptions = new HashSet<>();
+
+    @Override
+    public void subscribe(BankCard bankCard) {
+        users.add(bankCard.getUser());
+        subscriptions.add(new Subscription(bankCard.getNumber(), LocalDate.now()));
+    }
+
+    @Override
+    public Optional<Subscription> getSubscriptionByBankCardNumber(String cardNumber) {
+        return subscriptions.stream()
+                .filter(s -> s.getBankCardNumber().equals(cardNumber))
+                .findFirst();
+    }
+
+    @Override
+    public List<Subscription> getAllSubscriptionsByCondition(Predicate<Subscription> condition) {
+        return subscriptions.stream()
+                .filter(condition)
+                .collect(Collectors.toUnmodifiableList());
```

Comment:
Not an issue, but the ```.collect(Collectors.toUnmodifiableList())``` could be replaces with just ```toList()```. The task description might be a bit outdated